.github/workflows/run_tests.yml

```diff
@@ -120,3 +119,27 @@ jobs:
           cache: "poetry"
       - name: Run setup.py build
         run: python setup.py build
+
+  integration_test:
+    name: Run Integration Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9"]
```

Comment:
not fully sure but is it needed to specify a python version here? (can we extend this to >=3.9 as in dspy-ai?) and how does this affect users who don't have 3.9 - (saw a followup to this [issue here](https://github.com/stanfordnlp/dspy/issues/802) more so for dspy-ai but similar impact).